tools/delete_candidate.php

```diff
@@ -26,264 +41,192 @@
+        $DB->delete("participant_status", array("CandID" => $CandID));
+
+        //delete from the participant_status_history table
+        $DB->delete("participant_status_history", array("CandID" => $CandID));
+
+        //delete from parameter_candidate
+        $DB->delete("parameter_candidate", array("CandID" => $CandID));
+
+        //delete from candidate
+        $DB->delete("candidate", array("CandID" => $CandID));
+    } elseif ($printToSQL) {
+        echo "Generating all DELETE statements for CandID: " . $CandID . "And PSCID: " .
+            $PSCID . "\n";
+
+        //delete from the participant_status table
+        _printResultsSQL("participant_status", array("CandID" => $CandID), $output, $DB);
+
+        //delete from the participant_status_history table
+        _printResultsSQL("participant_status_history", array("CandID" => $CandID), $output, $DB);
+
+        //delete from parameter_candidate
+        _printResultsSQL("parameter_candidate", array("CandID" => $CandID), $output, $DB);
+
+        //delete from candidate
+        _printResultsSQL("candidate", array("CandID" => $CandID), $output, $DB);
+
+        _exportSQL($output, $CandID);
+    }
 }
-echo "----------------------delete each instrument table entry-------\n";
 
-//Delete from the feedback related tables
-$Feedbackids = $DB->pselect(
-    "SELECT fbt.FeedbackID from feedback_bvl_thread fbt WHERE CandID =:cid",
-    array('cid'=>$DCCID)
-);
-foreach ($Feedbackids as $Feedbackid) {
-    $DB->delete(
-        "feedback_bvl_entry", array('FeedbackID'=>$Feedbackid['FeedbackID'])
-    );
+function _printResultsSQL($table, $where, &$output, $DB)
+{
+    $query  = "DELETE FROM $table WHERE ";
+    $where  = $DB->_implodeWithKeys(' AND ', $where);
+    $query .= $where;
+    $query .= ";\n";
+
+    $output .=$query;
 }
 
-$DB->delete("feedback_bvl_thread", array('CandID'=>$DCCID));
-echo "----------------------delete feedback_bvl_thread---------------\n";
-
-//delete from the participant_status table
-$DB->delete("participant_status", array("CandID" => $DCCID));
-echo "----------------------delete participant_status----------------\n";
-//delete from the SNP_candidate_rel table
-$DB->delete("SNP_candidate_rel", array("CandID" => $DCCID));
-echo "----------------------delete SNP_candidate_rel-----------------\n";
-//delete from the participant_status_history table
-$DB->delete("participant_status_history", array("CandID" => $DCCID));
-echo "----------------------delete participant_status_history--------\n";
-//delete from candidate
-$DB->delete("candidate", array("CandID" => $DCCID));
-echo "----------------------delete candidate-------------------------\n";
-?>
+function _exportSQL ($output, $CandID) {
+    //export file
+    $filename = __DIR__ . "/../project/tables_sql/DELETE_candidate_$CandID.sql";
+    $fp       = fopen($filename, "w");
+    fwrite($fp, $output);
+    fclose($fp);
+}
+
+if ($confirm === false && $printToSQL === false) {
```

Comment:
add `&& !empty(sessions)`. Currently this message is a little strange when there are no sessions to be deleted and the script offers to delete the non-existent sessions.